reporter/html.js

```diff
@@ -329,44 +353,106 @@ function toolbarLooseFilter() {
+		dropDownHtml = "";
 
-	if ( !modulesList.length ) {
+	if ( !config.modules.length ) {
 		return false;
 	}
 
-	moduleFilterHtml += "<label for='qunit-modulefilter'>Module: </label>" +
-		"<select id='qunit-modulefilter' name='modulefilter'><option value='' " +
-		( QUnit.urlParams.module === undefined ? "selected='selected'" : "" ) +
-		">< All Modules ></option>";
-
-	for ( i = 0; i < modulesList.length; i++ ) {
-		moduleFilterHtml += "<option value='" +
-			escapeText( encodeURIComponent( modulesList[ i ] ) ) + "' " +
-			( QUnit.urlParams.module === modulesList[ i ] ? "selected='selected'" : "" ) +
-			">" + escapeText( modulesList[ i ] ) + "</option>";
+	for ( i = 0; i < config.modules.length; i++ ) {
+		if ( config.modules[ i ].name !== "" ) {
+			dropDownHtml += "<li><input id='" + config.modules[ i ].moduleId +
+			"' type='checkbox'" +
+			( config.moduleId.indexOf( config.modules[ i ].moduleId ) > -1 ? " checked" : "" ) +
+			"><label for='" + config.modules[ i ].moduleId + "'>" +
+			escapeText( config.modules[ i ].name ) + "</label></li>";
+		}
 	}
-	moduleFilterHtml += "</select>";
-
-	return moduleFilterHtml;
+	return dropDownHtml;
 }
 
-function toolbarModuleFilter() {
-	var toolbar = id( "qunit-testrunner-toolbar" ),
+function toolbarModuleFilter () {
+	var dropDownHtml = moduleDropDownHtml(),
+		toolbar = id( "qunit-testrunner-toolbar" ),
+		label = document.createElement( "label" ),
 		moduleFilter = document.createElement( "span" ),
-		moduleFilterHtml = toolbarModuleFilterHtml();
+		moduleSearch = document.createElement( "input" ),
+		dropDownContainer = document.createElement( "div" ),
+		dropDownList = document.createElement( "ul" ),
+		clearFilter = document.createElement( "li" );
 
-	if ( !toolbar || !moduleFilterHtml ) {
+	if ( !toolbar || !dropDownHtml ) {
 		return false;
 	}
 
-	moduleFilter.setAttribute( "id", "qunit-modulefilter-container" );
-	moduleFilter.innerHTML = moduleFilterHtml;
+	moduleSearch.placeholder = "Select modules";
 
-	addEvent( moduleFilter.lastChild, "change", applyUrlParams );
+	label.innerHTML = "Module: ";
+
+	clearFilter.innerHTML = "< All modules ><hr/>";
+
+	clearFilter.onclick = function() {
+		var i,
+			modulesList = dropDownList.getElementsByTagName( "input" );
+		for ( i = 0; i < modulesList.length; i++ )  {
+			modulesList[ i ].checked = false;
+		}
+		applyUrlParams();
+	};
+
+	moduleFilter.id = "qunit-modulefilter-container";
+	dropDownList.id = "qunit-modulefilter-dropdown-list";
+	moduleSearch.id = "qunit-modulefilter-search";
+	dropDownContainer.id = "qunit-modulefilter-dropdown-container";
+
+	dropDownContainer.style.display = "none";
+
+	dropDownList.appendChild( clearFilter );
+	moduleFilter.appendChild( label );
+	moduleFilter.appendChild( moduleSearch );
+	moduleFilter.appendChild( dropDownContainer ) ;
+
+	// Enables show/hide for the dropdown
+	addEvent( moduleSearch, "focus", function() {
+		var dropDownList = id( "qunit-modulefilter-dropdown-container" );
```

Comment:
Throughout this callback and the next one below, is there a reason to not use the `dropDownContainer` reference you already have in scope?